src/vfs/virtualFileSystemWithWatch.ts

```diff
@@ -0,0 +1,865 @@
+/* @internal */
+namespace ts.VirtualFS {
+    export interface VirtualServerHostCreationParameters {
+        useCaseSensitiveFileNames?: boolean;
+        executingFilePath: string;
+        currentDirectory?: string;
+        newLine?: string;
+        windowsStyleRoot?: string;
+        environmentVariables?: ESMap<string, string>;
+        runWithoutRecursiveWatches?: boolean;
+        runWithFallbackPolling?: boolean;

```

Comment:
These should be test only and moved to test version instead?

Reply:
I tried moving them, but it resulted in a lot of code duplication: TestServerHost.fsWatch and TestServerHost.constructor both copy their originals in VirtualServerHost. I think it's better to leave some test code in VirtualServerHost and avoid duplication.

Reply:
Never mind, it turns out almost nothing can be moved to TestServerHost without introducing some duplication and overriding some methods. I'll do that to move this and other properties.